simulariumio/file_converter.py

```diff
@@ -19,24 +21,207 @@
 class FileConverter(TrajectoryConverter):
     def __init__(self, input_file: InputFileData):
         """
-        This object loads the data in .simularium JSON format
-        at the input file
+        This object loads data from the input file in .simularium format.
 
         Parameters
         ----------
         input_file: InputFileData
-            A InputFileData object containing a string path
-            or string contents for the .simularium JSON file to load
+            A InputFileData object containing .simularium data to load
         """
-        print("Reading Simularium JSON -------------")
-        buffer_data = json.loads(input_file.get_contents())
+        try:
+            print("Reading Simularium JSON -------------")
+            buffer_data = json.loads(input_file.get_contents())
+        except UnicodeDecodeError:
+            print("Reading Simularium binary -------------")
+            buffer_data = FileConverter._load_binary(input_file)
         if (
             int(buffer_data["trajectoryInfo"]["version"])
             < CURRENT_VERSION.TRAJECTORY_INFO
         ):
-            buffer_data = self.update_trajectory_info_version(buffer_data)
+            buffer_data = FileConverter.update_trajectory_info_version(buffer_data)
         self._data = TrajectoryData.from_buffer_data(buffer_data)
 
+    @staticmethod
+    def _read_binary_file(input_file: InputFileData) -> Dict[str, Any]:
+        """
+        Read a .simularium binary file and return multiple views of the data
+        """
+        result = {}
+        with open(input_file.file_path, "rb") as open_binary_file:
+            result["byte_view"] = open_binary_file.read()
+            result["n_values"] = int(
```

Comment:
is this `n_values` actually used anywhere outside of here?  the file might contain encoded text data which does't really match up with BYTES_PER_VALUE.  Also why should this result be a dict rather than a namedTuple or some simple dataclass?

Reply:
also wondering if struct.unpack copies the data?

Reply:
see this as possible alternative to the use of struct.unpack in this case:  https://numpy.org/doc/stable/reference/generated/numpy.frombuffer.html

Reply:
I made a dataclass `BinaryFileData` and switched to using `numpy.frombuffer()` 🎉